<a href="https://colab.research.google.com/github/sairaawahid/Data-Science-in-Python-Data-Prep-and-EDA/blob/main/Exploratory%20Data%20Analysis/Exploring_Data_Analysis_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment: Exploring Data

**Dataset:**
1.   happiness_survey_data.csv

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)

Hi,

The marketing team would like to share out the five happiest countries of the 2010s on social media.
I’ve attached a notebook that another data scientist started with happiness data inside. I would recommend:
* Creating a list of each country’s highest happiness score, and then sorting it from happiest to least happy country
* Creating a list of each country’s average happiness score, and then sorting it from happiest to least happy country

Are there any differences between the two lists?

Thanks!\
Anna

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## Read in the data

In [ ]:
# import the data

import pandas as pd

df = pd.read_csv('/content/drive/MyDrive/Data Science in Pyhton by Maven Analytics/1. Data Prep & EDA/Data/happiness_survey_data.csv')
df.head()

,country_name,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
0,Afghanistan,2008,3.723590,0.450662,0.718114,50.500000
1,Afghanistan,2009,4.401778,0.552308,0.678896,50.799999
2,Afghanistan,2010,4.758381,0.539075,0.600127,51.099998
3,Afghanistan,2011,3.831719,0.521104,0.495901,51.400002
4,Afghanistan,2012,3.782938,0.520637,0.530935,51.700001


In [ ]:
# There are 2089 rows and 6 columns of data
df.shape

(2089, 6)

In [ ]:
# The years range from 2005 - 2021
# Happiness scores range from 2 - 8
df.describe()

,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
count,2089.000000,2089.000000,2076.000000,2057.000000,2031.000000
mean,2013.727621,5.473747,0.811542,0.745462,63.180326
std,4.455614,1.115567,0.118935,0.140751,6.948546
min,2005.000000,2.178809,0.290184,0.257534,6.720000
25%,2010.000000,4.651972,0.747664,0.651689,58.965000
50%,2014.000000,5.405246,0.834770,0.767357,64.980003
75%,2017.000000,6.294282,0.904682,0.857677,68.362499
max,2021.000000,8.018934,0.987343,0.985178,74.349998


![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)

## Explore the data

In [ ]:
# Filter out any data before 2010 and after 2019
mask = (df.year >= 2010) & (df.year < 2020)
filtered_df = df[mask]
filtered_df.head()

,country_name,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
2,Afghanistan,2010,4.758381,0.539075,0.600127,51.099998
3,Afghanistan,2011,3.831719,0.521104,0.495901,51.400002
4,Afghanistan,2012,3.782938,0.520637,0.530935,51.700001
5,Afghanistan,2013,3.572100,0.483552,0.577955,52.000000
6,Afghanistan,2014,3.130896,0.525568,0.508514,52.299999


In [ ]:
# Group the data by country and calculate the maximum happiness score for each one
filtered_df.groupby('country_name')['happiness_score'].max().reset_index()

,country_name,happiness_score
0,Afghanistan,4.758381
1,Albania,5.867422
2,Algeria,6.354898
3,Angola,5.589001
4,Argentina,6.775805
...,...,...
159,Venezuela,7.478455
160,Vietnam,5.767344
161,Yemen,4.350313
162,Zambia,5.243996


In [ ]:
# Sort the grouped countries by happiness score and return the top five
(filtered_df.groupby('country_name')['happiness_score']
            .max()
            .sort_values(ascending=False)
            .head(5))

,happiness_score
country_name,
Finland,7.858107
Denmark,7.788232
Switzerland,7.776209
Norway,7.678277
Canada,7.650346


In [ ]:
# Group the data by country and calculate the average happiness score for each one
filtered_df.groupby('country_name')['happiness_score'].mean()

,happiness_score
country_name,
Afghanistan,3.501017
Albania,4.976791
Algeria,5.389717
Angola,4.420299
Argentina,6.398067
...,...
Venezuela,5.858085
Vietnam,5.278218
Yemen,3.765910


In [ ]:
# Sort the grouped countries by happiness score and return the top five
(filtered_df.groupby('country_name')['happiness_score']
            .mean()
            .sort_values(ascending=False)
            .head(5))

,happiness_score
country_name,
Denmark,7.618252
Switzerland,7.568010
Finland,7.553138
Norway,7.541094
Iceland,7.518146


## Compare the two lists

### Happiest countries for social media
    
**Countries with the highest happiness scores**
1. Finland: 7.858107
2. Denmark: 7.788232
3. Switzerland: 7.776209
4. Norway: 7.678277
5. Canada: 7.650346

**Countries with the highest average happiness scores**
1. Denmark: 7.618252
2. Switzerland: 7.568010
3. Finland: 7.553138
4. Norway: 7.541094
5. Iceland: 7.518146

In [ ]:
# Compare the two lists
filtered_df[filtered_df.country_name.isin(['Denmark', 'Switzerland', 'Finland'])]

,country_name,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
481,Denmark,2010,7.770515,0.974977,0.943631,69.400002
482,Denmark,2011,7.788232,0.961736,0.934760,69.620003
483,Denmark,2012,7.519909,0.951437,0.932628,69.839996
484,Denmark,2013,7.588607,0.964708,0.920255,70.059998
485,Denmark,2014,7.507559,0.956344,0.941572,70.279999
486,Denmark,2015,7.514425,0.959701,0.941436,70.500000
487,Denmark,2016,7.557783,0.954452,0.948231,70.625000
488,Denmark,2017,7.593702,0.952100,0.955416,70.750000
489,Denmark,2018,7.648786,0.958219,0.935438,70.875000
490,Denmark,2019,7.693003,0.957706,0.963318,71.000000


![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)